x-pack/plugins/apm/public/components/shared/KueryBar/Typeahead/index.js

```diff
@@ -20,204 +21,218 @@ const KEY_CODES = {
   TAB: 9
 };
 
-export class Typeahead extends Component {
-  state = {
-    isSuggestionsVisible: false,
-    index: null,
-    value: '',
-    inputIsPristine: true
-  };
-
-  static getDerivedStateFromProps(props, state) {
-    if (state.inputIsPristine && props.initialValue) {
-      return {
-        value: props.initialValue
-      };
+export const Typeahead = injectI18n(
```

Comment:
@chandlerprall can you confirm something for me? I think this is the pattern that swallows up the name of the component in snapshots (and possibly also in React DevTools), but I want to make sure I'm remembering that right? 

If so, I think we need to unfortunately do `const Typeahead = // etc` and then beneath the component, do `export { Typeahead }` as a slightly strange way around that JS quirk.

Reply:
Although I do see a static `displayName` being set below so perhaps that's enough to solve it for this case.

Reply:
If the `displayName` doesn't work out, you can also name the class and it should propagate through as the component name:

`injectI18n(class Foo extends Comnponent { ... })`